src/UI/Shared/WB.UI.Shared.Enumerator/Services/MapService.cs

```diff
@@ -173,9 +179,29 @@ public void MoveTempMapToPermanent(string mapName)
             if (!this.fileSystemAccessor.IsFileExists(tempFileName))
                 return;
 
-            var newName = this.fileSystemAccessor.ChangeExtension(tempFileName, null);
-
-            this.fileSystemAccessor.MoveFile(tempFileName, newName);
+            var fileExtension = fileSystemAccessor.GetFileExtension(mapName);
+            if (fileExtension == ".shp") // shape file package
+            {
+                var mapFolder = this.fileSystemAccessor.CombinePath(GetMapsLocationOrThrow(), mapName);
+                try
+                {
+                    if (!fileSystemAccessor.IsDirectoryExists(mapFolder))
+                        fileSystemAccessor.CreateDirectory(mapFolder);
+                    archiveUtils.Unzip(tempFileName, mapFolder);

```

Comment:
I think before uzipping the target directory should be removed. Otherwise there existing folder content could affect the new one,